chain-indexer/src/domain/ledger_state.rs

```diff
@@ -132,7 +150,13 @@ impl LedgerState {
             created_unshielded_utxos,
             spent_unshielded_utxos,
             ledger_events,
-        } = self.apply_regular_transaction(&transaction.raw, block_parent_hash, block_timestamp)?;
+        } = self
+            .apply_regular_transaction(&transaction.raw, block_parent_hash, block_timestamp)
+            .map_err(|source| Error::ApplyTransaction {
+                variant: "regular".to_string(),
+                hash: format!("{:?}", transaction.hash),
```

Comment:
Here and in a LOT places you use `Debug`, but you should use `Display` instead, because `Debug` of our byte types shortens the output; for errors we want the full hash/address/etc. 
Please remove the format!("{:?}", ...) here and everywhere.